### DISCUSSION TITLE
Convert model to ONNX

### DISCUSSION THREAD

**Discussion by alueft000** [pyg-team/pytorch_geometric#4587] [General]
Hi, I'm trying to deploy the torch geometric models with ONNX. However, when I try to export the model to ONNX model, I get the error: `terminate called after throwing an instance of "pyblind11:error_already_set" What (): AssertionError.`

I looked through previous Github issues, but I am still not sure how to properly convert and run the model with ONNX for deployment. Would be great if the developing team or community provides a comprehensive and clear guide for deployment with ONNX. Thanks!

**Comment by rusty1s** [comment]
ONNX support is not provided at the moment. Currently, we recommend using TorchScript for this, see [here](https://github.com/pyg-team/pytorch_geometric/tree/master/examples/cpp).

**Comment by mdbenito** [comment]
For the record, some people seem to have had some success: https://github.com/pyg-team/pytorch_geometric/issues/5656. See also the discussion in https://github.com/pyg-team/pytorch_geometric/issues/728 and possibly others
